rosbag2_transport/src/rosbag2_transport/recorder.cpp

```diff
@@ -157,8 +157,8 @@ Recorder::get_requested_or_available_topics()
 
   std::regex topic_regex(record_options_.regex);
   std::regex exclude_regex(record_options_.exclude);
-  bool take = record_options_.all;
   for (const auto & kv : filtered_topics_and_types) {
+    bool take = record_options_.all;
     // regex_match returns false for 'empty' regex
     if (!record_options_.regex.empty()) {
```

Comment:
It would be nice to override `regex` by `record_options.all` in the logic as well.
```cpp
 if (!record_options.all && !record_options.regex.empty())
```
Please see my comment with explanation here https://github.com/ros2/rosbag2/pull/765#issuecomment-837263558